src/jwplayer/players/DesktopArticleVideoPlayer/DesktopArticleVideoPlayer.tsx

```diff
@@ -50,23 +57,35 @@ export const DesktopArticleVideoPlayerContent: React.FC<DesktopArticleVideoPlaye
 							isScrollPlayer ? styles.desktopArticleVideoWrapperScrollPlayer : styles.desktopArticleVideoWrapper
 						}
 					>
-						{adComplete && (
+						{adEngineComplete && jwpAdsSetupComplete && (
 							<div>
 								<div className={styles.topBar}>
 									{!isScrollPlayer && <UnmuteButton />}
 									{isScrollPlayer && (
 										<CloseButton dismiss={() => setDismissed(true)} iconColor={'#fff'} className={styles.closeButton} />
 									)}
 								</div>
-								<JwPlayerWrapper
-									getDismissed={getDismissed}
-									config={getArticleVideoConfig(videoDetails)}
-									onReady={(playerInstance) => {
-										articlePlayerOnReady(videoDetails, playerInstance);
-										setIsPlayerReady(true);
-									}}
-									stopAutoAdvanceOnExitViewport={false}
-								/>
+								{strategyRulesEnabled ? (
+									<JwPlayerWrapperWithStrategyRules
+										getDismissed={getDismissed}
+										config={getArticleVideoConfig(videoDetails)}
+										onReady={(playerInstance) => {
+											articlePlayerOnReady(videoDetails, playerInstance);
+											setIsPlayerReady(true);
+										}}
+										stopAutoAdvanceOnExitViewport={false}
+									/>
+								) : (
+									<JwPlayerWrapper
+										getDismissed={getDismissed}
+										config={getArticleVideoConfig(videoDetails)}
+										onReady={(playerInstance) => {
+											articlePlayerOnReady(videoDetails, playerInstance);
+											setIsPlayerReady(true);
+										}}
+										stopAutoAdvanceOnExitViewport={false}
```

Comment:
Since the `onReady` function is the same on lines 72 and 82, you could write it once above and pass it as a prop to the components.